spec/models/manageiq/providers/redhat/infra_manager/refresh/refresher_graph_target_host_spec.rb

```diff
@@ -0,0 +1,75 @@
+      allow_any_instance_of(@ovirt_service_inventory)
+        .to receive(:collect_vnic_profiles).and_return([])
+      @collector = ManageIQ::Providers::Redhat::Inventory::Collector
+      allow_any_instance_of(@collector)
+        .to receive(:collect_vnic_profiles).and_return([])
+      @ems.update_authentication(:default => {:userid => "admin@internal", :password => "123456"})
+      @ems.default_endpoint.verify_ssl = OpenSSL::SSL::VERIFY_NONE
+      allow(@ems).to receive(:supported_api_versions).and_return(%w(3 4))
+      stub_settings_merge(:ems => { :ems_redhat => { :use_ovirt_engine_sdk => true } })
+      allow(Spec::Support::OvirtSDK::ConnectionVCR).to receive(:new).and_call_original
+      allow(Spec::Support::OvirtSDK::ConnectionVCR).to receive(:new).with(kind_of(Hash)) do |opts|
+        Spec::Support::OvirtSDK::ConnectionVCR.new(opts,
+                                                   'spec/vcr_cassettes/manageiq/providers/redhat/infra_manager/refresh/ovirt_sdk_refresh_graph_target_host.yml',
+                                                   false)
+      end
+      stub_const("OvirtSDK4::Connection", Spec::Support::OvirtSDK::ConnectionVCR)
+    end
+
+    COUNTED_MODELS = [CustomAttribute, EmsFolder, EmsCluster, Datacenter].freeze
+
+    before(:each) do
+      @inventory_wrapper_class = ManageIQ::Providers::Redhat::InfraManager::Inventory::Strategies::V4
+
+      allow_any_instance_of(@inventory_wrapper_class).to(receive(:api).and_return("4.2.0_master."))
+      allow_any_instance_of(@inventory_wrapper_class).to(receive(:service)
+        .and_return(OpenStruct.new(:version_string => '4.2.0_master.')))
+    end
+
+    require 'yaml'
+    def load_response_mock_for(filename)
+      prefix = described_class.name.underscore
+      YAML.load_file(File.join('spec', 'models', prefix, 'response_yamls', filename + '.yml'))
+    end
+
+    it 'does not change the host when target refresh after full refresh' do
+      stub_settings_merge(:ems_refresh => { :rhevm => {:inventory_object_refresh => true }})
+      EmsRefresh.refresh(@ems)
+      @ems.reload
+      host = Host.where(:name => "host1").first
+      expect(host.ems_id).to eq(@ems.id)
+      saved_host = host_to_comparable_hash(host)
+      saved_counted_models = COUNTED_MODELS.map { |m| [m.name, m.count] }
+      saved_switches = host.switches.map { |switch| [switch.uid_ems, switch.name] }
+      expect(host.networks.count).to eq(2)
+      EmsRefresh.refresh(host)
+      host.reload
+      expect(host.networks.count).to eq(2)
+      expect(saved_switches).to match_array(host.switches.map { |switch| [switch.uid_ems, switch.name] })
+      EmsRefresh.refresh(host)
+      host.reload
+      expect([["00000000-0000-0000-0000-000000000009", "ovirtmgmt"], ["5c42817f-03fb-460e-a3ab-a8770553aeee", "vlan123t"]]).to match_array(host.switches.map { |switch| [switch.uid_ems, switch.name] })
```

Comment:
I know match array should be position independent but not sure how it works with nested arrays like this.  Maybe safer to check that `host.switches.map(&:uid_ems).match_array(["00000000-0000-0000-0000-000000000009", "5c42817f-03fb-460e-a3ab-a8770553aeee"])` then do `host.switches.detect { |switch| switch.uid_ems == "5c42817f-03fb-460e-a3ab-a8770553aeee"}` to check the names?

Reply:
hmm perhaps you are right - What if I use compoesed matchers instead?